### PR TITLE
Update samples for Bluetooth central

### PR DESCRIPTION
**Pull request by AdrianSoundy**
## Description

Added 2 samples for Bluetooth Central/Client

- Simple Watcher (Central1)
- Data collection sample (Central2)

Also updated Sample 3 to use as a test client for Central2 by adding code to change temperatures.

The reference to the Bluetooth Assembly is currently hard coded to assembly path. This will need to be changed to reference assembly on your system.

Once a NuGet is available these samples will be updated in this PR. 

## Motivation and Context
Required for new Bluetooth Central updates

## Types of changes

- [x] Improvement (non-breaking change that improves a sample)
- [ ] Bug fix (fixes an issue with a current sample)
- [x] New Sample (adds a new sample)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Signed-off-by: AdrianSoundy <AdrianSoundy@gmail.com>

### CODE REVIEW

File: samples/Bluetooth/BluetoothLESample3/BluetoothLESample3.nfproj [hunk 1]
```diff
@@ -33,9 +33,14 @@
       <HintPath>packages\nanoFramework.CoreLibrary.1.12.0\lib\mscorlib.dll</HintPath>
       <Private>True</Private>
     </Reference>
+<<<<<<< Updated upstream
```
Comment (josesimoes) [inline samples/Bluetooth/BluetoothLESample3/BluetoothLESample3.nfproj:36]: Something went wrong with the merge... 😯

File: samples/Bluetooth/BluetoothLESample3/Program.cs [hunk 2]
```diff
@@ -134,6 +134,37 @@ public static void Main()
...
+                {
+                    t1 += 1.3F;
+                    t3 += 2.1F;
+
+                    EnvService.UpdateValue(iTempOut, t1);
+                    EnvService.UpdateValue(iTempOutMin, t3);
+                    Thread.Sleep(5000);
+                }
+
+                // Up
```
Comment (josesimoes) [inline samples/Bluetooth/BluetoothLESample3/Program.cs:155]: Maybe _down_ ?

File: samples/Bluetooth/BluetoothLESample3/packages.config [hunk 3]
```diff
@@ -1,8 +1,14 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="nanoFramework.CoreLibrary" version="1.12.0" targetFramework="netnanoframework10" />
+<<<<<<< Updated upstream
```
Comment (josesimoes) [inline samples/Bluetooth/BluetoothLESample3/packages.config:4]: Here too

File: samples/Bluetooth/BluetoothLESample3/BluetoothLESample3.nfproj (hunk 1 above)
Comment (AdrianSoundy) [reply samples/Bluetooth/BluetoothLESample3/BluetoothLESample3.nfproj]: Don't know what wrong. Thought it was working project. Fixed now.

File: samples/Bluetooth/BluetoothLESample3/packages.config (hunk 3 above)
Comment (AdrianSoundy) [reply samples/Bluetooth/BluetoothLESample3/packages.config]: Fixed

File: samples/Bluetooth/BluetoothLESample3/Program.cs (hunk 2 above)
Comment (AdrianSoundy) [reply samples/Bluetooth/BluetoothLESample3/Program.cs]: Yes down

**Review by josesimoes — APPROVED** [review]
LGTM!

**Pull request merged**